test/room_test.rb

```diff
@@ -0,0 +1,105 @@
+        @room = Hotel::Room.new(id: 34, reservations: [])
+      end
+      
+    it "is an instance of room" do
+        expect(@room).must_be_kind_of Hotel::Room
+    end
+
+    it "returns the right id" do
+        expect(@room.id).must_equal 34
+    end
+
+    it "returns the right reservations" do
+        expect(@room.reservations).must_equal []
+    end
+
+    it "can return a list of rooms" do
+        rooms = Hotel::Room.list_of_rooms
+        expect(rooms[0].id).must_equal 1
+        expect(rooms[0].reservations).must_be_kind_of Array
+    end
+
+    it "can return a list of reservations for a room on a certain date" do
+        start_date= DateTime.new(2020,1,3.5)
+        end_date = DateTime.new(2020,2,4.5)
+        start_date1= DateTime.new(2020,2,4.5)
+        end_date1 = DateTime.new(2020,2,29.5)
+        start_date2= DateTime.new(2020,1,3.5)
+        end_date2 = DateTime.new(2020,1,4.5)
+        start_date3= DateTime.new(2020,2,3.5)
+        end_date3 = DateTime.new(2020,2,4.5)
+        # puts @hotel.rooms[0].list_of_reservations_for_data_range(start_date,end_date)
+        # puts @hotel.rooms[0].list_of_reservations_for_data_range(start_date1,end_date1)
+        # puts @hotel.rooms[0].list_of_reservations_for_data_range(start_date2,end_date2)
+        expect(@hotel.rooms[0].list_of_reservations_for_data_range(start_date,end_date)).must_equal [@reservation1]
+        expect(@hotel.rooms[0].list_of_reservations_for_data_range(start_date1,end_date1)).must_equal [@reservation1,@reservation2,@reservation3]
+        expect(@hotel.rooms[0].list_of_reservations_for_data_range(start_date2,end_date2)).must_equal []
+        expect(@hotel.rooms[0].list_of_reservations_for_data_range(start_date3,end_date3)).must_equal [@reservation1]
+    end
+
+    it "can return an reservation in a room on a specific date" do
+        date = DateTime.new(2020,2,4.5)
+        date1 = DateTime.new(2020,3,4.5)
+        date2 = DateTime.new(2020,2,5.5)
+        date3 = DateTime.new(2020,2,14.5)
+        expect(@hotel.rooms[0].specific_date_reservations(date)).must_equal [@reservation1]
+        expect(@hotel.rooms[0].specific_date_reservations(date1)).must_equal []
+        expect(@hotel.rooms[0].specific_date_reservations(date2)).must_equal []
+        expect(@hotel.rooms[0].specific_date_reservations(date3)).must_equal [@reservation3]
+    end
+
+    it "returns false when a room is anavailable" do
+        date = DateTime.new(2020,2,4.5)
+        date1 = DateTime.new(2020,3,4.5)
+        date2 = DateTime.new(2020,2,5.5)
+        date3 = DateTime.new(2020,2,14.5)
+        expect(@hotel.rooms[0].specific_date_availability(date)).must_equal [false]
+        expect(@hotel.rooms[0].specific_date_availability(date1)).must_equal []
+        expect(@hotel.rooms[0].specific_date_availability(date2)).must_equal []
+        expect(@hotel.rooms[0].specific_date_availability(date3)).must_equal [false]
+    end
```

Comment:
There are a lot of ways that date ranges can overlap! Somewhere you need to exhaustively test them to make sure your date checking logic is correct.